@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -852,12 +867,15 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
 
                 try {
                     // Triangulate the footprint polyline.
-                    const triangles = earcut(points, holes, stride);
+                    // For sphere projection the stride of Ring.contour and Ring.points can be
+                    // different b/c Ring.contour contains texture coordinates where Ring.points
+                    // does not.
+                    const triangles = earcut(points, holes, isSpherical ? 3 : stride);
 
-                    if (this.m_decodeInfo.projection.type === ProjectionType.Spherical) {
+                    if (isSpherical) {
                         const geom = new THREE.Geometry();
 
-                        for (let i = 0; i < vertices.length; i += 3) {
+                        for (let i = 0; i < vertices.length; i += stride) {
```

Comment:
to earcut you pass in a stride of `3` if `isSpherical`, but here you used `stride`, I don't quite follow.

Reply:
For Mercator Ring.point and Ring.contour are the same (i.e. Ring.point and Ring.contour have a stride of 5 if texture coordinates are enabled.
For Globe Ring.point contains the points projected with equirectangular projection and Ring.contour the points projected with the sphere projection. The reason is that we want to do the triangulation in 2d but the projected points are 3d (i.e. points is only passed to earcut and used nowhere else).
This is all a bit messy and will hopefully be refactored soon.